src/hypothesis/strategies.py

```diff
@@ -174,16 +174,18 @@ def one_of(*args):
         except TypeError:
             pass
 
+    strategies = []
```

Comment:
Hmm. Then I'm confused, but that suggests that something funky is happening elsewhere too. In particular it probably means that strategy definitions that use | internally (like floats()) aren't getting flattened.

Looking at this, I think that's because you no longer have branches() on the various wrapper types.

Related question: Does 'x | y | z' get flattened in the same way that one_of(x, y, z) does? I think it doesn't right now.